dracut/30disk-uuid/disk-uuid.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+cmdline=( $(</proc/cmdline) )
+cmdline_arg() {
+    local name="$1" value="$2"
+    for arg in "${cmdline[@]}"; do
+        if [[ "${arg%%=*}" == "${name}" ]]; then
+            value="${arg#*=}"
+        fi
+    done
+    echo "${value}"
+}
+
+# First, check for the cmdline option in case it was set.
+RANDOMIZE_DISK_GUID=$(cmdline_arg flatcar.randomize_disk_guid)
+
+# But we are no longer detecting the non-randomized UUID in GRUB,
+# so set the base guid here if the cmdline was not set.
+BASE_GUID="00000000-0000-0000-0000-000000000001"
+if [[ -z "$RANDOMIZE_DISK_GUID" ]]; then
+    RANDOMIZE_DISK_GUID="${BASE_GUID}"
+fi
+
+# Check that the by-diskuuid path exists. If it doesn't, this script is
+# useless, so highlight that by exiting with a failure code
+DEVPATH="/dev/disk/by-diskuuid"
+if [[ ! -e ${DEVPATH} ]]; then
+    echo "${DEVPATH} does not exist. Bailing out"
+    exit 1
+fi
+
+# Verify the device actually exists before trying to touch it
+DEVICE="${DEVPATH}/${RANDOMIZE_DISK_GUID}"
+if [[ -e "${DEVICE}" ]]; then
+    /usr/bin/cgpt repair ${DEVICE} && \
+    /usr/sbin/sgdisk --disk-guid=R ${DEVICE} && \
+    /usr/bin/udevadm settle
```

Comment:
If the goal is to have the `/dev/disk/by-diskuuid/000…` entry go away and a new `/dev/disk/by-diskuuid/` entry to appear, it would make sense to wait for this to happen. The udevadm settle command is more like a `sleep 1` and not a reliable waiter. Maybe it's not needed and then it could be removed but if it is needed, I would prefer a real waiter.

Reply:
I don't know what "a real waiter" would look like. But this part of the code comes from the previous unit and I'd rather not change that in this commit, since that part was working well enough already.

Reply:
It was there but through the reordering it is also done at a different time which could uncover a race condition.
The waiter could first count the number of entries in `/dev/disk/by-diskuuid/`, and then wait for `/dev/disk/by-diskuuid/000…` to not exist but the number of entries be at least the same as before.

Reply:
Let's leave it as long as it works but I think it would be cleaner to remove it and see if it still works, and if not we know the condition that we want to wait for.